src/server/run_mir.cpp

```diff
@@ -110,7 +221,20 @@ void mir::run_mir(
             {
                 for (auto sig : intercepted)
                 {
-                    old_handler[sig] = signal(sig, fatal_signal_cleanup);
+                    struct sigaction sig_handler_desc;
+                    sig_handler_desc.sa_flags = SA_SIGINFO;
+                    sig_handler_desc.sa_sigaction = &fatal_signal_cleanup;
+
+                    if (sigaction(sig, &sig_handler_desc, old_handlers.initialiser_for_signal(sig)))
```

Comment:
This lifetime of temporary trick with `AtomicOutPtr` is quite cool.

Reply:
Yeah! It was really interesting seeing the summary of P1132R8 scroll across my twitter just before I had a need for such a gadget!